src/de.rs

```diff
@@ -103,3 +103,16 @@ impl<'de> Deserialize<'de> for Box<Bytes> {
         Ok(bytes.into())
     }
 }
+
+impl<'de, T> Deserialize<'de> for Option<T>
+where
+    T: Deserialize<'de> + serde::de::Deserialize<'de> + From<&'de [u8]>,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let bytes: Option<&[u8]> = serde::Deserialize::deserialize(deserializer)?;
```

Comment:
This only works for deserializers that support deserializing &[u8], not for deserializers that support only owned bytes. For example it wouldn't work for Bincode when deserializing using [`bincode::deserialize_from`](https://docs.rs/bincode/1.2.1/bincode/fn.deserialize_from.html) because there isn't a way to borrow bytes from a std::io::Read impl.